### PR TITLE
fix roi pooling

### PR DESCRIPTION
**Pull request by XingHongChenIntel**
Need to merge ( https://github.com/Flex-plaidml-team/openvino/pull/33 ) first.
change the `max` and `bilinear` mode to accord with `ngraph`.
comment the `FP16` type, we don't support it now.

### CODE REVIEW

File: plaidml/bridge/openvino/ops/roi_pooling.cpp [hunk 1]
```diff
@@ -120,17 +156,13 @@ void registerROIPooling() {
...
       if (method == "max") {
         // translate ROI coordinates from their input normalize scale to feature map scale.
-        for (int i = 1; i < coord.size(); i++) {
+        for (int i = 0; i < coord.size(); i++) {
           coord[i] = std::round(coord[i] * spatial_ratio);
         }
-        pooled_tensor = crop_max_pooling(slice_I, coord, pooled_height, pooled_width);
+        pooled_tensor = crop_max_pooling(slice_I, coord, pooled_height, pooled_width, height, width);
       } else if (method == "bilinear") {
         // follow ngraph implementation, which doesn't use spatial_ratio in "bilinear" method.
```
Comment (YangleiZouIntel) [inline plaidml/bridge/openvino/ops/roi_pooling.cpp:135]: This is good, should keep it as it is. Please discard my previous comment.

File: plaidml/bridge/openvino/ops/roi_pooling.cpp [hunk 2]
```diff
@@ -14,70 +14,109 @@ using namespace InferenceEngine;  // NOLINT[build/namespaces]
...
+  edsl::Tensor bottom_left = get_pieces(I, bottom_y_index, left_x_index);
+  edsl::Tensor bottom_right = get_pieces(I, bottom_y_index, right_x_index);
+
+  std::vector<edsl::TensorDim> dims(I.rank());
+  I.bind_dims(dims);
+  dims[2] = edsl::TensorDim(pooled_h);
+  dims[3] = edsl::TensorDim(pooled_w);
+  auto top = top_left + (top_right - top_left) * (in_x - left_x_index);
+  auto bottom = bottom_left + (bottom_right - bottom_left) * (in_x - left_x_index);
+  // output = top + (bottom - top) * (in_y - top_y_index), we have to reshape mul openrands.
```
Comment (YangleiZouIntel) [inline plaidml/bridge/openvino/ops/roi_pooling.cpp]: Probably it is better to add a new line before this comment.

Comment (YangleiZouIntel) [reply plaidml/bridge/openvino/ops/roi_pooling.cpp]: Typo: operands

File: plaidml/bridge/openvino/ops/roi_pooling.cpp [hunk 3]
```diff
@@ -14,70 +14,109 @@ using namespace InferenceEngine;  // NOLINT[build/namespaces]
...
+      // if start equal to end, we have to guarantee that index is at least one. and they are not over the border.
+      auto slice_w_index = end_w - start_w > 0 ? end_w - start_w : 1;
+      auto slice_h_index = end_h - start_h > 0 ? end_h - start_h : 1;
+      auto h_index = start_h >= height ? height - 1 : start_h;
+      auto w_index = start_w >= width ? width - 1 : start_w;
+      auto h_tensor = edsl::index({edsl::TensorDim(slice_h_index)}, 0) + h_index;
+      auto w_tensor = edsl::index({edsl::TensorDim(slice_w_index)}, 0) + w_index;
+      auto gather_w = edsl::gather(I, w_tensor).axis(3);
+      auto crop = static_cast<edsl::Tensor>(edsl::gather(gather_w, h_tensor).axis(2));
       // get max value from bin, then put it to pooled tensor.
```
Comment (YangleiZouIntel) [inline plaidml/bridge/openvino/ops/roi_pooling.cpp:57]: Please add new line here before the comment.

**Comment by YangleiZouIntel** [comment]
This looks very good. But it seems very slow on CPU, some tests take 5-15 seconds, I will check if it can be optimized.

File: plaidml/bridge/openvino/ops/roi_pooling.cpp (hunk 2 above)
Comment (XingHongChenIntel) [reply plaidml/bridge/openvino/ops/roi_pooling.cpp]: done

File: plaidml/bridge/openvino/ops/roi_pooling.cpp (hunk 3 above)
Comment (XingHongChenIntel) [reply plaidml/bridge/openvino/ops/roi_pooling.cpp:57]: done

**Comment by YangleiZouIntel** [comment]
Submitted in upstream.

**Pull request closed without merging**